test/TestUtilities/Test.Utility/TestDotnetCLiUtility.cs

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.Protocol;
+using NuGet.Test.Utility;
+using NuGet.Versioning;
+
+namespace NuGet.Test.Utility
+{
+    public static class TestDotnetCLiUtility
+    {
+        internal static string SdkVersion { get; private set; }
+        internal static NuGetFramework SdkTfm { get; private set; }
+        internal static string CliDirSource { get; private set; }
+        internal static string SdkDirSource { get; private set; }
+
+
+        public static TestDirectory CopyAndPatchLatestDotnetCli(string sdkVersion = null, string sdkTfm = null)
```

Comment:
I think this file will be easier to understand if you stop making the two parameters optional, and have the dotnet integration tests pass in appropriate values.

Reply:
I tried to refactor the code to pass something for both dotnet integration tests and crossframwork verify tests.
It doesn't look good though. 
Since for dotnet integration tests, the `sdkversion `and `sdktfm` could be dynamically determined without passing any arguments, if I tried to pass something, and keep the methods of getting `sdkversion `and `sdktfm ` in TestDotnetCLiUtility, there will be some duplicate code of the preparation.
So I tried to add some comments for `CopyAndPatchLatestDotnetCli` and hope it will be helpful.
Pls let me know if it's good with you, or if I raise a follow up issue to refactor this. Thanks!